features/roommembermoderation/impl/src/main/kotlin/io/element/android/features/roommembermoderation/impl/RoomMemberModerationPresenter.kt

```diff
@@ -0,0 +1,213 @@
+                        }
+                    }
+                }
+                is InternalRoomMemberModerationEvents.DoKickUser -> {
+                    selectedUser?.let {
+                        coroutineScope.kickUser(it.userId, event.reason, kickUserAsyncAction)
+                    }
+                    selectedUser = null
+                }
+                is InternalRoomMemberModerationEvents.DoBanUser -> {
+                    selectedUser?.let {
+                        coroutineScope.banUser(it.userId, event.reason, banUserAsyncAction)
+                    }
+                    selectedUser = null
+                }
+                is InternalRoomMemberModerationEvents.DoUnbanUser -> {
+                    selectedUser?.let {
+                        coroutineScope.unbanUser(it.userId, unbanUserAsyncAction)
+                    }
+                    selectedUser = null
+                }
+                is InternalRoomMemberModerationEvents.Reset -> {
+                    selectedUser = null
+                    kickUserAsyncAction.value = AsyncAction.Uninitialized
+                    banUserAsyncAction.value = AsyncAction.Uninitialized
+                    unbanUserAsyncAction.value = AsyncAction.Uninitialized
+                }
+            }
+        }
+
+        return InternalRoomMemberModerationState(
+            canKick = canKick.value,
+            canBan = canBan.value,
+            selectedUser = selectedUser,
+            actions = moderationActions.value,
+            kickUserAsyncAction = kickUserAsyncAction.value,
+            banUserAsyncAction = banUserAsyncAction.value,
+            unbanUserAsyncAction = unbanUserAsyncAction.value,
+            eventSink = { handleEvent(it) },
+        )
+    }
+
+    private fun computeModerationActions(
+        member: RoomMember?,
+        canKick: Boolean,
+        canBan: Boolean,
+        currentUserMemberPowerLevel: Long,
+    ): PersistentList<ModerationActionState> {
+        return buildList {
+            add(ModerationActionState(action = ModerationAction.DisplayProfile, isEnabled = true))
+            // Assume the member is a regular user when it's unknown
+            val targetMemberPowerLevel = member?.powerLevel ?: 0
+            val canModerateThisUser = currentUserMemberPowerLevel > targetMemberPowerLevel
+            // Assume the member is joined when it's unknown
+            val membership = member?.membership ?: RoomMembershipState.JOIN
+            if (canKick) {
+                val isKickEnabled = canModerateThisUser && membership.isActive()
+                add(ModerationActionState(action = ModerationAction.KickUser, isEnabled = isKickEnabled))
+            }
+            if (canBan) {
+                if (membership == RoomMembershipState.BAN) {
+                    add(ModerationActionState(action = ModerationAction.UnbanUser, isEnabled = canModerateThisUser))
```

Comment:
Sorry, I have forgotten: the code coverage is decreasing a bit, maybe cover this line by a unit test?